# HG changeset patch
# Parent 7f1176cf80338d7f795722d67f13a269aa4476b2
# User Frédéric Wang <fred.wang@free.fr>
Always start the lookup of largeop variants at size=1 (bug 584332). r=karlt

diff --git a/layout/mathml/mathfontSTIXSizeOneSym.properties b/layout/mathml/mathfontSTIXSizeOneSym.properties
--- a/layout/mathml/mathfontSTIXSizeOneSym.properties
+++ b/layout/mathml/mathfontSTIXSizeOneSym.properties
@@ -44,18 +44,17 @@
 # can be rendered with STIXSize* set of fonts,
 # with some help from STIXNonUnicode, STIXGeneral and STIXIntegralsD.
 
 external.1 = STIXNonUnicode
 external.2 = STIXSizeTwoSym
 external.3 = STIXSizeThreeSym
 external.4 = STIXSizeFourSym
 external.5 = STIXSizeFiveSym
-external.6 = STIXGeneral
-external.7 = STIXIntegralsD
+external.6 = STIXIntegralsD
 
 #        [ T/L |  M  | B/R |  G  | size0 ... size{N-1} ]
 \u0028 = \u239B\uFFFD\u239D\u239C\uFFFD((@2(@3(@4 # (
 \u0029 = \u239E\uFFFD\u23A0\u239F\uFFFD))@2)@3)@4 # )
 \u005B = \u23A1\uFFFD\u23A3\u23A2\u005B[[@2[@3[@4 # [
 \u005D = \u23A4\uFFFD\u23A6\u23A5\u005D]]@2]@3]@4 # ]
 \u007B = \u23A7\u23A8\u23A9\u23AA\u007B{{@2{@3{@4 # {
 \u007D = \u23AB\u23AC\u23AD\u23AA\u007D}}@2}@3}@4 # }
@@ -79,43 +78,43 @@ external.7 = STIXIntegralsD
 \u2A09 = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A09 # N-ARY TIMES OPERATOR
 \u2AFF = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2AFF # N-ARY WHITE VERTICAL BAR
 
 # E000 stix-radical symbol vertical extender
 # E001 stix-radical symbol top corner
 \u221A = \uE001@1\uFFFD\u221A@4\uE000@1\uFFFD\u221A\u221A@2\u221A@3 # Sqrt, radic
 
 # Integrals
-\u222B = \u2320\uFFFD\u2321\u23AE\uFFFD\u222B@6\u222B@7
-\u222C = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u222C@6\u222C@7
-\u222D = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u222D@6\u222D@7
-\u222E = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u222E@6\u222E@7
-\u222F = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u222F@6\u222F@7
-\u2230 = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2230@6\u2230@7
-\u2231 = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2231@6\u2231@7
-\u2232 = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2232@6\u2232@7
-\u2233 = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2233@6\u2233@7
-\u2A0B = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A0B@6\u2A0B@7
-\u2A0C = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A0C@6\u2A0C@7
-\u2A0D = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A0D@6\u2A0D@7
-\u2A0E = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A0E@6\u2A0E@7
-\u2A0F = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A0F@6\u2A0F@7
-\u2A10 = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A10@6\u2A10@7
-\u2A11 = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A11@6\u2A11@7
-\u2A12 = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A12@6\u2A12@7
-\u2A13 = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A13@6\u2A13@7
-\u2A14 = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A14@6\u2A14@7
-\u2A15 = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A15@6\u2A15@7
-\u2A16 = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A16@6\u2A16@7
-\u2A17 = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A17@6\u2A17@7
-\u2A18 = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A18@6\u2A18@7
-\u2A19 = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A19@6\u2A19@7
-\u2A1A = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A1A@6\u2A1A@7
-\u2A1B = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A1B@6\u2A1B@7
-\u2A1C = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A1C@6\u2A1C@7
+\u222B = \u2320\uFFFD\u2321\u23AE\uFFFD\u222B@6
+\u222C = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u222C@6
+\u222D = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u222D@6
+\u222E = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u222E@6
+\u222F = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u222F@6
+\u2230 = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2230@6
+\u2231 = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2231@6
+\u2232 = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2232@6
+\u2233 = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2233@6
+\u2A0B = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A0B@6
+\u2A0C = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A0C@6
+\u2A0D = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A0D@6
+\u2A0E = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A0E@6
+\u2A0F = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A0F@6
+\u2A10 = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A10@6
+\u2A11 = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A11@6
+\u2A12 = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A12@6
+\u2A13 = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A13@6
+\u2A14 = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A14@6
+\u2A15 = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A15@6
+\u2A16 = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A16@6
+\u2A17 = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A17@6
+\u2A18 = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A18@6
+\u2A19 = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A19@6
+\u2A1A = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A1A@6
+\u2A1B = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A1B@6
+\u2A1C = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u2A1C@6
 
 \u27E8 = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u27E8\u27E8@2\u27E8@3\u27E8@4 # LeftAngleBracket
 \u27E9 = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u27E9\u27E9@2\u27E9@3\u27E9@4 # RightAngleBracket
 
 \u23DE = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u23DE\u23DE@2\u23DE@3\u23DE@4\u23DE@5 # &OverBrace; (Unicode)
 \uFE37 = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u23DE\u23DE@2\u23DE@3\u23DE@4\u23DE@5 # &OverBrace; (MathML 2.0)
 \u23B4 = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u23B4\u23B4@2\u23B4@3\u23B4@4\u23B4@5 # &OverBracket;
 \u23DC = \uFFFD\uFFFD\uFFFD\uFFFD\uFFFD\u23DC\u23DC@2\u23DC@3\u23DC@4\u23DC@5 # &OverParenthesis; (Unicode)
diff --git a/layout/mathml/nsMathMLChar.cpp b/layout/mathml/nsMathMLChar.cpp
--- a/layout/mathml/nsMathMLChar.cpp
+++ b/layout/mathml/nsMathMLChar.cpp
@@ -1219,21 +1219,18 @@ nsMathMLChar::StretchEnumContext::TryVar
     largeop && (NS_STRETCH_VARIABLE_MASK & mStretchHint) == 0;
   PRBool maxWidth = (NS_STRETCH_MAXWIDTH & mStretchHint) != 0;
 
   nscoord bestSize =
     isVertical ? mBoundingMetrics.ascent + mBoundingMetrics.descent
                : mBoundingMetrics.rightBearing - mBoundingMetrics.leftBearing;
   PRBool haveBetter = PR_FALSE;
 
-  // figure out the starting size : if this is a largeop, start at 2 else 1
-  PRInt32 size = 1; // size=0 is the char at its normal size
-  if (largeop && aGlyphTable->BigOf(mPresContext, mChar, 2).Exists()) {
-    size = 2;
-  }
+  // start at size = 1 (size = 0 is the char at its normal size)
+  PRInt32 size = 1;
 #ifdef NOISY_SEARCH
   printf("  searching in %s ...\n",
            NS_LossyConvertUTF16toASCII(aFamily).get());
 #endif
 
   nsGlyphCode ch;
   while ((ch = aGlyphTable->BigOf(mPresContext, mChar, size)).Exists()) {
 
